src/react/utils.js

```diff
@@ -843,21 +843,16 @@ export function sanitizeReactElementForFirstRenderOnly(realm: Realm, reactElemen
   let typeValue = Get(realm, reactElement, "type");
 
   // ensure ref is null, as we don't use that on first render
-  Properties.Set(realm, reactElement, "ref", realm.intrinsics.null, false);
+  setProperty(realm, reactElement, "ref", realm.intrinsics.null);
   // when dealing with host nodes, we want to sanitize them futher
   if (typeValue instanceof StringValue) {
     let propsValue = Get(realm, reactElement, "props");
     if (propsValue instanceof ObjectValue) {
       // remove all values apart from string/number/boolean
       for (let [propName] of propsValue.properties) {
-        invariant(propsValue instanceof ObjectValue);
-        let value = getProperty(realm, propsValue, propName);
-
-        // skip children and style
-        if (propName === "children" || propName === "style") {
-          continue;
-        }
-        if (!(value instanceof StringValue || value instanceof NumberValue || value instanceof BooleanValue)) {
+        // check for onSomething prop event handlers, i.e. onClick
+        if (propName.length > 2 && propName[0] === "o" && propName[1] === "n") {
```

Comment:
I think checking `.toLowerCase()` for both letters would match what React does more closely.